application/src/main/java/org/togetherjava/tjbot/logging/discord/DiscordLogForwarder.java

```diff
@@ -160,22 +162,25 @@ private List<LogMessage> validateBatch(List<LogMessage> logBatch) {
     private record LogMessage(WebhookEmbed embed,
             Instant timestamp) implements Comparable<LogMessage> {
 
-        private static LogMessage ofEvent(LogEvent event) {
+        private static final String BASE_PACKAGE = "org.togetherjava.tjbot.";
+
+        private static LogMessage ofEvent(LogEvent event, String sourceCodeBaseUrl) {
             String authorName = event.getLoggerName();
+            String authorUrl =
+                    linkToSource(event.getSource().getClassName(), sourceCodeBaseUrl).orElse(null);
```

Comment:
What if `event.getSource().getClassName()` is null?

Reply:
its a NotNull method

Reply:
I haven't looked deep into it's source but from the looks of it it uses `StackTraceElement` which can throw an NPE. Is this supressed by log4j or is there some fall back?

Reply:
I'm not sure if the getSource can return null,
the StackTraceElement#getClassName() is not null though
![image](https://github.com/Together-Java/TJ-Bot/assets/88111627/22d8f74b-e0a0-4ddd-927b-d1700ffd1a33)

this declaringClass gets initialized with this:
```
this.declaringClass  = Objects.requireNonNull(declaringClass, "Declaring class is null");
```
and I will hope that it will not get changed via reflection lol

Reply:
`Objects.requireNonNull(declaringClass, "Declaring class is null");` so the "Declaring class is null" is actually what the contents of the NPE will be.

https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#requireNonNull-T-java.lang.String-

Reply:
yeah but I am not calling the constructor, I am just calling getSource which returns an already existing StackTraceElement, no?

Reply:
hm the AbstractLogEvent returns null by default
💀 
![image](https://github.com/Together-Java/TJ-Bot/assets/88111627/3c229a03-29de-40c1-b4b7-66f5edd5ee08)

Reply:
It's a small change, in your `linkToSource` just add a null check :) i.e.

`if (source == null) return Optional.empty();

Reply:
![image](https://github.com/Together-Java/TJ-Bot/assets/88111627/10157563-f090-4730-bb44-9ba03658692b)

Reply:
bruh

Reply:
oh mb, we have ParametersAreNonnullByDefault
need to add nullable

Reply:
Do the check on this then: `event.getSource().getClassName()` e.g. 

```String className = event.getSource().getClassName();
String authorName = className != null ? linkToSource(...): "some fallback text";```